android/src/main/java/com/epicshaggy/biometric/NativeBiometric.java

```diff
@@ -156,408 +151,455 @@ public void isAvailable(PluginCall call) {
-                    call.reject("Failed to save credentials", e);
-                    e.printStackTrace();
+                    call.reject("Failed to get credentials", e);
                 } catch (IOException e) {
-                    call.reject("Failed to save credentials", e);
-                    e.printStackTrace();
+                    call.reject("Failed to get credentials", e);
                 }
             } else {
-                call.reject("Missing properties");
+                call.reject("No credentials found");
             }
+        } else {
+            call.reject("No server name was provided");
         }
+    }
+
+    @PluginMethod
+    public void signData(final PluginCall call)  {
+        Intent intent = new Intent(getContext(), AuthSignActivity.class);
+
+        intent.putExtra("challengeString", call.getString("challengeString"));
+        intent.putExtra("title", "Authenticate");
+
+        bridge.saveCall(call);
+        startActivityForResult(call, intent, "verifySignResult");
+    }
+
+    @ActivityCallback
+    private void verifySignResult(PluginCall call, ActivityResult result) {
+        JSObject ret = new JSObject();
 
-        @PluginMethod()
-        public void getCredentials ( final PluginCall call){
-            String KEY_ALIAS = call.getString("server", null);
-
-            SharedPreferences sharedPreferences = getContext().getSharedPreferences(NATIVE_BIOMETRIC_SHARED_PREFERENCES, Context.MODE_PRIVATE);
-            String username = sharedPreferences.getString("username", null);
-            String password = sharedPreferences.getString("password", null);
-            if (KEY_ALIAS != null) {
-                if (username != null && password != null) {
-                    try {
-                        JSObject jsObject = new JSObject();
-                        jsObject.put("username", decryptString(username, KEY_ALIAS));
-                        jsObject.put("password", decryptString(password, KEY_ALIAS));
-                        call.resolve(jsObject);
-                    } catch (GeneralSecurityException e) {
-                        call.reject("Failed to get credentials", e);
-                    } catch (IOException e) {
-                        call.reject("Failed to get credentials", e);
-                    }
-                } else {
-                    call.reject("No credentials found");
+        if (result.getResultCode() == Activity.RESULT_OK) {
+            Intent data = result.getData();
+            if (data.hasExtra("result")) {
+                switch (data.getStringExtra("result")) {
+                    case "success":
+                        ret.put("signedData", data.getStringExtra("signedData"));
+                        call.resolve(ret);
+                        break;
+                    case "authenticationError":
+                        call.reject(data.getStringExtra("errorDetails"), data.getStringExtra("errorCode"));
```

Comment:
Question, call reject here means it will throw error ya?

Reply:
Yep, it will throw error which in js side can be found in catch block.

Reply:
Okay, I like this behaviour. Thank you :+1: